src/types.ts

```diff
@@ -257,52 +264,144 @@ export interface FinalizationData {
     finalizers: PartyInfo[];
 }
 
-export interface ExchangeRate {
+export interface Ratio {
     numerator: bigint;
     denominator: bigint;
 }
 
+export type ExchangeRate = Ratio;
+
+export interface InclusiveRange<N extends number> {
+    min: N;
+    max: N;
+}
+
+/**
+ * Used as index of a reward period, or as a number of reward periods.
+ * A reward period spans a number of epochs, as defined in chain parameters as "rewardPeriodLength"
+ */
+export type RewardPeriod = bigint;
+/** Index of an epoch, or number of epochs. */
+export type Epoch = bigint;
```

Comment:
Do we actually use these aliases to signify index?

Reply:
The comment is basically taken from the haskell implementation of the corresponding type